doc/devel/development.md

```diff
@@ -0,0 +1,1932 @@
+in the component, the transaction ID in this component will be used as the correlator.
+
+During the duration of a transaction, all the log entries created by the code will write the current Transaction ID and
+correlator for the operation being executed.
+
+### Library overview
+
+In order to use the library, add the following dependency to your package.json file:
+
+```json
+"iotagent-node-lib": "*"
+```
+
+In order to use this library, first you must require it:
+
+```javascript
+var iotagentLib = require('iotagent-node-lib');
+```
+
+The library supports four groups of features, one for each direction of the communication: client-to-server and
+server-to-client (and each flow both for the client and the server). Each feature set is defined in the following
+sections.
+
+### Function reference
+
+The following fucntions are available in the library:
+
+-   [iotagentLib.activate()](#iotagentlibactivate)
+-   [iotagentLib.deactivate()](#iotagentlibdeactivate)
+-   [iotagentLib.register()](#iotagentlibregister)
+-   [iotagentLib.unregister()](#iotagentlibunregister)
+-   [iotagentLib.update()](#iotagentlibupdate)
+-   [iotagentLib.setCommandResult()](#iotagentlibsetcommandresult)
+-   [iotagentLib.listDevices()](#iotagentliblistdevices)
+-   [iotagentLib.setDataUpdateHandler()](#iotagentlibsetdataupdatehandler)
+-   [iotagentLib.setDataQueryHandler()](#iotagentlibsetdataqueryhandler)
+-   [iotagentLib.setNotificationHandler()](#iotagentlibsetnotificationhandler)
+-   [iotagentLib.setCommandHandler()](#iotagentlibsetcommandhandler)
+-   [iotagentLib.setMergePatchHandler()](#iotagentlibsetmergepatchhandler)
+-   [iotagentLib.setProvisioningHandler()](#iotagentlibsetprovisioninghandler)
+-   [iotagentLib.setRemoveDeviceHandler()](#iotagentlibsetremovedevicehandler)
+-   [iotagentLib.setConfigurationHandler()](#iotagentlibsetconfigurationhandler)
+-   [iotagentLib.setRemoveConfigurationHandler()](#iotagentlibsetremoveconfigurationhandler)
+-   [iotagentLib.getDevice()](#iotagentlibgetdevice)
+-   [iotagentLib.getDeviceByName()](#iotagentlibgetdevicebyname)
+-   [iotagentLib.getDevicesByAttribute()](#iotagentlibgetdevicesbyattribute)
+-   [iotagentLib.retrieveDevice()](#iotagentlibretrievedevice)
+-   [iotagentLib.mergeDeviceWithConfiguration()](#iotagentlibmergedevicewithconfiguration)
+-   [iotagentLib.getConfiguration()](#iotagentlibgetconfiguration)
+-   [iotagentLib.findConfiguration()](#iotagentlibfindconfiguration)
+-   [iotagentLib.getEffectiveApiKey()](#iotagentlibgeteffectiveapikey)
+-   [iotagentLib.subscribe()](#iotagentlibsubscribe)
+-   [iotagentLib.unsubscribe()](#iotagentlibunsubscribe)
+-   [iotagentLib.ensureSouthboundDomain()](#iotagentlibensuresouthbounddomain)
+-   [iotagentLib.finishSouthBoundTransaction()](#iotagentlibfinishsouthboundtransaction)
+-   [iotagentLib.startServer()](#iotagentlibstartserver)
+-   [iotagentLib.request()](#iotagentlibrequest)
```

Comment:
Is this function list up to date? Maybe some has been removed, some has been added, some has change its signature of functionality, etc.

The task of reviewing it in deep can be hard, but al least we should add a WARNING note in the documentation, explaining that it may be obsolete.

Reply:
Added warning note here: https://github.com/telefonicaid/iotagent-node-lib/pull/1371/commits/e46c18c1072247932b22ace41306a0c301588c23

Reply:
This is going to be done in a subsequent PR. Noted here: https://github.com/telefonicaid/iotagent-node-lib/issues/1329#issue-1643605738
NTC